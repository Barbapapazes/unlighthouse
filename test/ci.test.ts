import { resolve, join } from 'path'
import fs from 'fs-extra'
import { expect, it, beforeAll, describe, afterAll } from 'vitest'
import { execa } from 'execa'

export const cacheDir = resolve(__dirname, '.cache')
export const ci = resolve(__dirname, '../packages/unlighthouse/bin/unlighthouse-ci.cjs')

beforeAll(async() => {
  await fs.remove(cacheDir)
})

afterAll(async() => {
  await fs.remove(cacheDir)
})

describe('ci', () => {
  it('tests harlanzw.com', async() => {
    const { output } = await runCli(resolve(__dirname, 'fixtures/harlanzw.config.ts'))

    expect(output).toMatchSnapshot()
  })
})

async function runCli(configFileFixture: string) {
  const testDir = resolve(cacheDir, Date.now().toString())

  await fs.ensureDir(testDir)

  const config = await fs.readFile(configFileFixture, 'utf8')
  await fs.writeFile(join(testDir, 'unlighthouse.config.ts'), config)

  const { exitCode, stdout, stderr } = await execa('node', [ci, '--root', testDir, '--debug', '--host', 'harlanzw.com'], {
    cwd: testDir,
  })

  const logs = stdout + stderr
  console.log(logs)
  if (exitCode !== 0)
    throw new Error(logs)

  const output = await fs.readJson(resolve(testDir, '.lighthouse', 'ci-result.json'))

  return {
    output,
    logs,
  }
}
