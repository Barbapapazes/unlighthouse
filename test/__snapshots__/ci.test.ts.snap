// Vitest Snapshot v1

exports[`ci > tests harlanzw.com 1`] = `
[
  {
    "path": "/",
    "score": 0.97,
  },
  {
    "path": "/about/",
    "score": 0.98,
  },
  {
    "path": "/blog/how-the-heck-does-vite-work/",
    "score": 0.93,
  },
  {
    "path": "/blog/scale-your-vue-components/",
    "score": 0.92,
  },
  {
    "path": "/blog/vue-automatic-component-imports/",
    "score": 0.97,
  },
  {
    "path": "/contact/",
    "score": 0.98,
  },
  {
    "path": "/open-blogging/",
    "score": 0.98,
  },
  {
    "path": "/subscribe/",
    "score": 0.98,
  },
]
`;
